be/src/olap/single_replica_compaction.h

```diff
@@ -37,6 +37,8 @@ class SingleReplicaCompaction final : public CompactionMixin {
     Status prepare_compact() override;
     Status execute_compact() override;
 
+    inline CompactionType real_compact_type() const { return _compaction_type; }
```

Comment:
why not just compaction_type() ?

Reply:
'compact_type()' has been used  to return ReaderType.